org.eclipse.xtext/src/org/eclipse/xtext/validation/DiagnosticConverterImpl.java

```diff
@@ -58,7 +58,7 @@ public void convertResourceDiagnostic(Diagnostic diagnostic, Severity severity,
 		issue.setSeverity(severity);
 		issue.setLineNumber(diagnostic.getLine());
 		issue.setColumn(diagnostic.getColumn());
-		issue.setMessage(diagnostic.getMessage());
```

Comment:
why for message only and not for all other properties here?

Reply:
We do have API to customize the diagnostic messages. Why is this necessary here?

Reply:
Could you give me a pointer? Might make this one obsolete. I did not came across when solving our requirement.

Reply:
diagnostic#data carries an Exception instance (actually it is promoting NPEs that are normally swallowed by `org.eclipse.xtext.validation.AbstractDeclarativeValidator.MethodWrapper.handleInvocationTargetException(Throwable, State)`).
The exception's class name and stack trace is put into the issue message (customer wants to see this in the editor).
So instead of "Error in EValidator" the exception is taken from the diagnostic's data.

Reply:
So you want to bug the user with programming errors the dsl developer does?

Reply:
Me? No, the customer requires that. Yes, they want it that way.

Reply:
Still don’t know why you need to change both places

Reply:
1) First throws the exception => gets into the `data` of the diagnostic
2) The exception data object should be translated to the issue's message

Reply:
Where is the place for 1

Reply:
custom MethodWrapper that does not swallow the NPE!

Reply:
*custom*!

Reply:
not here!

Reply:
Yes but why can’t then the other thing produce a proper message

Reply:
Because it should _not_ produce a specific message, but just handle the NPE as any other runtime exception. And the converter here is responsible to _convert_ the diagnostic to an issue.

Reply:
are we still talking about a simple getter here?

Reply:
so should that be the place to be protected?

Reply:
Might be an alternative.